<a href="https://colab.research.google.com/github/maryandy2002/maryandy2002/blob/main/MetodoPuntoFijo.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Método de Punto Fijo
Utilizaremos el método de punto fijo para encontrar la solución de la ecuación $x^3+4x^2-10=0$ la cual tiene una raíz única en $[1,2]$



En este bloque importamos las librerías que se van a utilizar, $tabulate$ para crear una tabla al final para visualizar los resultados y $math$ para codificar la función que usaremos para aproximar la raíz

In [3]:
from tabulate import tabulate as tb
import math as mt

Existen muchas formas para cambiar la ecuación que estamos trabajando a la forma de punto fijo $x=f(x)$, en éste codigo utilizaremos la siguiente: $x=g(x)=\sqrt{\frac{10}{4+x}}$, la cuál esta codificada a continuación con ayuda de la librerira $math$

In [14]:
def funcion(x):
  return mt.sqrt(10/(4+x))

Aqui se definen nuestras listas $n$ para ir contando la cantidad de iteraciones, $e$ para ir calculando el error aproximado y $p$ que es donde iremos guardando las aproximaciones a la raíz que vayamos obteniendo

In [17]:
n=[0]
e=[0]
p=[1.5]

En este bloque se llevan a cabo las iteraciones del método de punto fijo, el cuál, termina cuando se ha alcanzado un error menor a $10^7$

In [20]:
while (True):
  p.append(funcion(p[-1]))
  e.append(abs(p[-1]-p[-2]))
  if (e[-1]<0.0000001):
    break
  n.append(n[-1]+1)

Finalmente se realiza la tabla correspondiente al procedimiento que se acaba de realizar utilizando el método de punto fijo

In [21]:
table=[n,p,e]
print(tb((zip(*table)),headers=["n","p","e"]))

  n        p            e
---  -------  -----------
  0  1.5      0
  1  1.3484   0.1516
  2  1.36738  0.0189766
  3  1.36496  0.00241936
  4  1.36526  0.000307733
  5  1.36523  3.9154e-05
  6  1.36523  4.98151e-06


Concluimos que la raíz para la ecuación $x^3+4x^2-10=0$ es $x=1.36523$ con un error $\varepsilon=4.9815\times10^{-6}$